src/function/table/system/duckdb_prepared_statements.cpp

```diff
@@ -0,0 +1,130 @@
+#include "duckdb/function/table/system_functions.hpp"
+
+#include "duckdb/common/exception.hpp"
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/parser/parser.hpp"
+#include "duckdb/main/client_data.hpp"
+#include "duckdb/main/prepared_statement_data.hpp"
+
+namespace duckdb {
+
+struct DuckDBPreparedStatementsData : public GlobalTableFunctionState {
+	DuckDBPreparedStatementsData() : offset(0) {
+	}
+
+	vector<std::pair<string, shared_ptr<PreparedStatementData>>> entries;
+	idx_t offset;
+};
+
+static unique_ptr<FunctionData> DuckDBPreparedStatementsBind(ClientContext &context, TableFunctionBindInput &input,
+                                                             vector<LogicalType> &return_types, vector<string> &names) {
+	names.emplace_back("name");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	names.emplace_back("statement");
+	return_types.emplace_back(LogicalType::VARCHAR);
+
+	names.emplace_back("prepare_time");
+	return_types.emplace_back(LogicalType::TIMESTAMP);
```

Comment:
We don't need to have properties we don't support here - the `duckdb_XXX` functions are always subject to additions in the future. We can just introduce unsupported elements in the view that we create for `pg_prepared_statements` as constants

Reply:
Done 👍 

I used NULL for all the unsupported parameters, reason being:
- We don't store the `prepare_time`, I don't think we can retrieve this
- Same for `from_sql`, we don't keep a record of this
- `generic_plans` - not even sure what this is
- `custom_plans` - again, no clue what this is

Reply:
Related to this, which we dont store stats for:
https://dev.to/aws-heroes/are-postgresql-prepared-statements-really-prepared-4cm2